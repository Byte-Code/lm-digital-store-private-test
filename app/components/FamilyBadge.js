import React, { Component, PropTypes } from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import { Link } from 'react-router';
import styled from 'styled-components';

const Wrapper = styled.div`
  width: ${props => props.width}px;
  display: flex;
  flex-direction: column;
  margin-bottom: 20px;
  box-shadow: 0 0 8px 0 #e4e4e4;
`;

const ImageWrapper = styled.img`
  height: ${props => props.height}px;
`;

const NameWrapper = styled.div`
  width: 100%;
  background-color: #ffffff;
  padding: 15px 10px;
  height: 70px;
`;

const FamilyName = styled.div`
  height: 28px;
  font-family: LeroyMerlinSans Bold;
  font-size: 20px;
  line-height: 1.4;
  text-transform: uppercase;
`;

const ItemCount = styled.div`
  height: 20px;
  font-size: 16px;
  line-height: 1.4;
  font-family: LeroyMerlinSans Light-Italic;
`;

export default class FamilyBadge extends Component {
  static propTypes = {
    family: ImmutablePropTypes.map.isRequired,
    size: PropTypes.string.isRequired
  }

  getSize() {
    const { size } = this.props;

    switch (size) {
      case 'square-big':
      default:
        return { height: 490, width: 490 };
      case 'square-small':
        return { height: 235, width: 235 };
      case 'vertical':
        return { height: 560, width: 235 };
      case 'horizontal':
        return { height: 235, width: 490 };
    }
  }

  render() {
    const { family } = this.props;

    const familyName = family.get('familyName');
    const image = family.get('image');
    const itemCount = family.get('itemCount');
    const categoryCode = family.get('categoryCode');
    const { height, width } = this.getSize();

    return (
      <Link to={`/catalogue/${categoryCode}`}>
        <Wrapper width={width}>
          <ImageWrapper
            src={image}
            alt="alt1"
            width={width}
            height={height}
          />
          <NameWrapper>
            <FamilyName>{familyName}</FamilyName>
            <ItemCount>{itemCount} prodotti</ItemCount>
          </NameWrapper>
        </Wrapper>
      </Link>
    );
  }
}
