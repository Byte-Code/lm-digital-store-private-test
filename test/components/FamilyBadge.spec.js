import React from 'react';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';

import FamilyBadge, { Wrapper, ImageWrapper } from '../../app/components/FamilyBadge';

describe('FamilyBadge', () => {
  const family = fromJS({
    familyName: 'family',
    image: 'image',
    categoryCode: '123'
  });

  it('case square-big: should render with the right height and width', () => {
    const squareBig = shallow(
      <FamilyBadge family={family} size="square-big" />
    );
    expect(squareBig.find(Wrapper).prop('width')).toEqual(490);
    expect(squareBig.find(ImageWrapper).prop('height')).toEqual(561);
    expect(squareBig).toMatchSnapshot();
  });

  it('case square-small: should render with the right height and width', () => {
    const squareSmall = shallow(
      <FamilyBadge family={family} size="square-small" />
    );
    expect(squareSmall.find(Wrapper).prop('width')).toEqual(235);
    expect(squareSmall.find(ImageWrapper).prop('height')).toEqual(235);
    expect(squareSmall).toMatchSnapshot();
  });


  it('case horizontal: should render with the right height and width', () => {
    const horizontal = shallow(
      <FamilyBadge family={family} size="horizontal" />
    );
    expect(horizontal.find(Wrapper).prop('width')).toEqual(490);
    expect(horizontal.find(ImageWrapper).prop('height')).toEqual(235);
    expect(horizontal).toMatchSnapshot();
  });


  it('case wrong prop: should render with default height and width', () => {
    const defaultSize = shallow(
      <FamilyBadge family={family} size="defualt" />
    );
    expect(defaultSize.find(Wrapper).prop('width')).toEqual(490);
    expect(defaultSize.find(ImageWrapper).prop('height')).toEqual(235);
    expect(defaultSize).toMatchSnapshot();
  });
});
