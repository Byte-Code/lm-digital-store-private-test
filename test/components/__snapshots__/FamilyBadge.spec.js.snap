// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FamilyBadge case horizontal: should render with the right height and width 1`] = `
<Link
  onlyActiveOnIndex={false}
  style={Object {}}
  to="/catalogue/123"
>
  <styled.div
    width={490}
  >
    <styled.img
      alt="alt1"
      height={235}
      src="image"
      width={490}
    />
    <styled.div>
      <styled.div>
        family
      </styled.div>
    </styled.div>
  </styled.div>
</Link>
`;

exports[`FamilyBadge case square-big: should render with the right height and width 1`] = `
<Link
  onlyActiveOnIndex={false}
  style={Object {}}
  to="/catalogue/123"
>
  <styled.div
    width={490}
  >
    <styled.img
      alt="alt1"
      height={561}
      src="image"
      width={490}
    />
    <styled.div>
      <styled.div>
        family
      </styled.div>
    </styled.div>
  </styled.div>
</Link>
`;

exports[`FamilyBadge case square-small: should render with the right height and width 1`] = `
<Link
  onlyActiveOnIndex={false}
  style={Object {}}
  to="/catalogue/123"
>
  <styled.div
    width={235}
  >
    <styled.img
      alt="alt1"
      height={235}
      src="image"
      width={235}
    />
    <styled.div>
      <styled.div>
        family
      </styled.div>
    </styled.div>
  </styled.div>
</Link>
`;

exports[`FamilyBadge case wrong prop: should render with default height and width 1`] = `
<Link
  onlyActiveOnIndex={false}
  style={Object {}}
  to="/catalogue/123"
>
  <styled.div
    width={490}
  >
    <styled.img
      alt="alt1"
      height={235}
      src="image"
      width={490}
    />
    <styled.div>
      <styled.div>
        family
      </styled.div>
    </styled.div>
  </styled.div>
</Link>
`;
